rollup/internal/controller/watcher/batch_proposer.go

```diff
@@ -170,124 +188,82 @@ func (p *BatchProposer) proposeBatch() (*encoding.Batch, error) {
 		}
 	}
 
+	useCodecv0 := dbChunks[0].StartBlockNumber < p.banachForkHeight
+
 	daChunks, err := p.getDAChunks(dbChunks)
 	if err != nil {
-		return nil, err
+		return err
 	}
 
 	parentDBBatch, err := p.batchOrm.GetLatestBatch(p.ctx)
 	if err != nil {
-		return nil, err
+		return err
 	}
 
 	var batch encoding.Batch
-	if parentDBBatch != nil {
-		batch.Index = parentDBBatch.Index + 1
-		parentDABatch, err := codecv0.NewDABatchFromBytes(parentDBBatch.BatchHeader)
-		if err != nil {
-			return nil, err
-		}
-		batch.TotalL1MessagePoppedBefore = parentDABatch.TotalL1MessagePopped
-		batch.ParentBatchHash = common.HexToHash(parentDBBatch.Hash)
+	batch.Index = parentDBBatch.Index + 1
+	var parentDABatch *codecv0.DABatch
+	parentDABatch, err = codecv0.NewDABatchFromBytes(parentDBBatch.BatchHeader)
```

Comment:
We might need to use `codecv1` based on `parentDBBatch.StartBlockNumber`.

Reply:
nice catch. changed in [10ad42c](https://github.com/scroll-tech/scroll/pull/1201/commits/10ad42cf4d568f10ca21323dbd8675e42e73faeb)

Reply:
update: the above commit introduces a bug (and cannot pass the unit tests), which is fixed in later commits.